### PR TITLE
Create smart-descriptions.json

### PR DESCRIPTION
**Pull request by Men-hau**
For Vectra
Linked with #[39504](https://github.com/SekoiaLab/platform/issues/39504)

### CODE REVIEW

**Review by squioc — CHANGES_REQUESTED** [review]

File: Vectra/vectra_cognito_detect/_meta/smart-descriptions.json [hunk 1]
```diff
@@ -0,0 +1,14 @@
+[
+  {
+    "value": "{vectra.detection.name} logged on to {host.ip.0} with Detection type {vectra.detection.type} and Threat {vectra.risk_score_norm}",
+    "conditions": [
+    ],
+    "relationships": [
+      {
+        "source": "vectra.detection.name",
+        "target": "host_ip",
```
[the remark this answers is not in the document]
Comment (Men-hau) [reply Vectra/vectra_cognito_detect/_meta/smart-descriptions.json]: Yes
it is written `host_ip` in the message
`Host.ip` is the field equivalent parsed on our side

**Review by squioc — COMMENTED** [review]
In your case, only field existence is requested for smart-description conditions.
With values, you settle here, the smart-description will only match one extremely precise case of events.

**Comment by Men-hau** [comment]
new smart descriptions proposition based on [event samples documentation](https://docs.sekoia.io/xdr/features/collect/integrations/network/vectra/#event-samples)

File: Vectra/vectra_cognito_detect/_meta/smart-descriptions.json [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+      {
+         "field": "observer.ip",
+      },
+      {
+         "field": "vectra.detection.last_type",
+      }  
+    ],
+    "relationships": [
+      {
+        "source": "vectra.history.host_access.name",
```
Comment (squioc) [inline Vectra/vectra_cognito_detect/_meta/smart-descriptions.json:39]: To be used as an edge of a relationship, The declaration of the custom field should declare an observable property like [this one](https://github.com/SEKOIA-IO/intake-formats/blob/ce500dc94685f1c0b5fa7c5dc801fc4c7923585c/AWS/aws-cloudtrail/_meta/fields.yml#L90-L93)

File: Vectra/vectra_cognito_detect/_meta/smart-descriptions.json [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+      },
+      {
+         "field": "destination.ip",
+      }   
+    ],
+    "relationships": [
+      {
+        "source": "source.ip",
+        "target": "destination.ip",
+        "type": "vectra.detection.reason"
```
Comment (squioc) [inline Vectra/vectra_cognito_detect/_meta/smart-descriptions.json:95]: A field to be used in the property `type`, only text.

**Review by squioc — CHANGES_REQUESTED** [review]

File: Vectra/vectra_cognito_detect/_meta/smart-descriptions.json (hunk 3 above)
Comment (Men-hau) [reply Vectra/vectra_cognito_detect/_meta/smart-descriptions.json]: #457

Now line 43

File: Vectra/vectra_cognito_detect/_meta/smart-descriptions.json [hunk 4]
```diff
@@ -0,0 +1,230 @@
...
+        "field": "vectra.detection.name"
+      }
+    ],
+    "relationships": [
+      {
+        "source": "host.ip",
+        "target": "destination.ip",
+        "type": "set"
+      }
+    ]
```
Comment (PierrickV) [inline Vectra/vectra_cognito_detect/_meta/smart-descriptions.json:22]:
```suggestion
    "value": "{vectra.detection.name} detected on {host.ip}",
    "conditions": [
      {
        "field": "event.action",
        "value": "INFO"
      },
      {
        "field": "host.ip"
      },
      {
        "field": "vectra.detection.name"
      }
    ],
    "relationships": [
      {
        "source": "host.ip",
        "target": "destination.ip",
        "type": "connected to"
      }
    ]
```

File: Vectra/vectra_cognito_detect/_meta/smart-descriptions.json [hunk 5]
```diff
@@ -0,0 +1,230 @@
...
+  },
+  {
+    "value": "{observer.ip} logged in with {vectra.account.uid} on {vectra.history.host_access.name} and detected {vectra.detection.last_type}",
```
Comment (PierrickV) [inline Vectra/vectra_cognito_detect/_meta/smart-descriptions.json:25]:
```suggestion
    "value": "{observer.ip} logged in with {vectra.account.uid} on {vectra.history.host_access.name} and {vectra.detection.last_type} was detected",
```

File: Vectra/vectra_cognito_detect/_meta/smart-descriptions.json [hunk 6]
```diff
@@ -0,0 +1,230 @@
...
+    ]
+  },
+  {
+    "value": "Threat type {vectra.detection.last_type} detected with risk evaluated at {vectra.risk_score_norm} for connection between {observer.ip} and {host.ip}",
+    "conditions": [
+      {
+        "field": "observer.ip"
+      },
+      {
+        "field": "host.ip"
```
Comment (PierrickV) [inline Vectra/vectra_cognito_detect/_meta/smart-descriptions.json:55]:
```suggestion
    "value": "Threat type {vectra.detection.last_type} detected with risk evaluated at {vectra.risk_score_norm} for connection between {source.ip} and {destinaton.ip}",
    "conditions": [
      {
        "field": "source.ip"
      },
      {
        "field": "destination.ip"
```

File: Vectra/vectra_cognito_detect/_meta/smart-descriptions.json [hunk 7]
```diff
@@ -0,0 +1,230 @@
...
+    "relationships": [
+      {
+        "source": "observer.ip",
+        "target": "host.ip",
+        "type": "connected to"
+      }
+    ]
+  },
+  {
+    "value": "{action.name} with {vectra.detection.reason} for connection between {destination.ip} and {source.ip}",
```
Comment (PierrickV) [inline Vectra/vectra_cognito_detect/_meta/smart-descriptions.json:73]:
```suggestion
    "value": "{action.name} with {vectra.detection.reason} detected for connection between {source.ip} and {destination.ip}",
```

File: Vectra/vectra_cognito_detect/_meta/smart-descriptions.json [hunk 8]
```diff
@@ -0,0 +1,230 @@
...
+    "relationships": [
+      {
+        "source": "host.ip",
+        "target": "destination.ip",
+        "type": "connected to"
+      }
+    ]
+  },
+  {
+    "value": "{observer.ip} detected {event.action} and with risk evaluted as {vectra.risk_score_norm} on {host.ip}",
```
Comment (PierrickV) [inline Vectra/vectra_cognito_detect/_meta/smart-descriptions.json:145]:
```suggestion
    "value": "{observer.ip} detected {event.action} with risk evaluated as {vectra.risk_score_norm} on {host.ip}",
```

**Review by PierrickV — APPROVED** [review]

**Pull request merged**